src/Microsoft.ML.TimeSeries/SequentialTransformerBase.cs

```diff
@@ -191,7 +256,9 @@ public void ProcessWithoutBuffer(ref TInput input, ref TOutput output)
         internal readonly string OutputColumnName;
         private protected ColumnType OutputColumnType;
 
-        public bool IsRowToRowMapper => false;
+        public bool IsRowToRowMapper => true;
+
+        public TState StateRef{ get; set; }
```

Comment:
>public TState StateRef{ get; set; } [](start = 8, length = 35)

I don't see this as a good idea at all.
The whole cloning mechanism could be implemented completely internally, without making anything public, or mutable. Just don't use `MemberwiseClone`, and write it yourself. This also has the benefit of self-documenting what's being cloned and how.

Reply:
Sounds good.

---
In reply to: [236008749](https://github.com/dotnet/machinelearning/pull/1618#discussion_r236008749) [](ancestors = 236008749)